### PR TITLE
Add ArchiveExecutionTask to our set of tasks

### PR DESCRIPTION
**Pull request by MichaelSnowden**
**What changed?**
In this PR, I've added a new task category for archival tasks. In addition, I've added a new task type within that category for archiving workflow executions.

**Why?**
I made this change as part of a larger effort to improve our archival process. This will allow us to define an archival task processor which will guarantee archival before deletion.

**How did you test it?**
There is no actual task processing in this diff, just the definition of the tasks. I did verify that the serialize/deserialization process did return the right data, and I also verified the task category and type are correct.

**Potential risks**
If someone uses this, they might mistakenly think it will archive their data (the processors for these tasks aren't set up yet).

**Is hotfix candidate?**
Not a hot fix candidate.

### CODE REVIEW

File: service/history/tasks/category.go [hunk 1]
```diff
@@ -89,6 +91,12 @@ var (
 		cType: CategoryTypeImmediate,
 		name:  CategoryNameVisibility,
 	}
+
+	CategoryArchival = Category{
```
Comment (yycptt) [inline service/history/tasks/category.go:95]: Add this category to `categories` on L103 as well.

Comment (MichaelSnowden) [reply service/history/tasks/category.go:95]: Ah, thanks! Done

Comment (MichaelSnowden) [reply service/history/tasks/category.go:95]: Actually, adding this means we need to have a queue factory registered for this category, which I'm only planning on adding in a later diff, so I'd prefer to keep it out of this map for now.

File: proto/internal/temporal/server/api/persistence/v1/executions.proto [hunk 2]
```diff
@@ -225,6 +225,17 @@ message TimerTaskInfo {
...
 
+
+message ArchivalTaskInfo {
+    int64 task_id = 1;
+    string namespace_id = 2;
+    string workflow_id = 3;
+    string run_id = 4;
+    temporal.server.api.enums.v1.TaskType task_type = 5;
+    int64 version = 6;
+    google.protobuf.Timestamp visibility_time = 7 [(gogoproto.stdtime) = true];
```
Comment (yux0) [inline proto/internal/temporal/server/api/persistence/v1/executions.proto:236]: I think you need a field for retention time as well

**Review by yux0 — APPROVED** [review]

**Review by yycptt — APPROVED** [review]

**Pull request merged**